### DISCUSSION TITLE
Waiting a long time for data to be saved

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#11427] [Archive]
I have a collection in which I have 2 columns with M2M relations, When I already have several dozen selected records in M2M and I save the record in collection. The recording takes 20-25 seconds. Is this normal or could I have set something incorrectly? How to verify it?
The connection to the database is good (578ms Latency), and with ordinary records the saving takes a moment...

**Comment by rijkvanzanten** [comment]
> The connection to the database is good (578ms Latency),

Is this the connection speed from Directus to the database, or from your computer to Directus?? That's a very long time for a ping between Node and the DB if that's the case 🤔

**Reply by koksikus** [reply]
This is the information that Directus shows me. Today I have over 300... Can you give me some advice on what speeds are great and should reduce the record writing time from 20-25s 
![Zrzut ekranu 2022-03-03 070655](https://user-images.githubusercontent.com/90528895/156506607-d2abbbd4-3e15-4cc2-8321-f8f111c2ee06.png)
Thank you

**Reply by rijkvanzanten** [reply]
20-25 seconds for saving is unusually long. That being said, it's a little difficult for me to tell what might be the cause with the limited information available. Would you be able to share a database dump or schema snapshot file so we can debug and see if it's a problem in the Directus code itself?

**Reply by koksikus** [reply]
I don't know how to do database dump or schema snapshot. 
I noticed that if I make a change to a normal field it takes a second, if I delete an item from the M2M list it takes max 5 seconds. On the other hand, it takes the longest to add another record in M2M... 
I'm still using Directus 9.2.1 because in the newer version there is a known bug that characters appear only after saving.
![Zrzut ekranu 2022-03-17 123210](https://user-images.githubusercontent.com/90528895/158804909-471b22a2-70c6-4e81-bcaa-6e304044f7fe.jpg)

I took a screenshot today. I cleared before saving and all records are at the time of writing data. That 15 seconds of recording is the correct time?
